bio/bio.go

```diff
@@ -0,0 +1,23 @@
+/*
+Package bio provides utilities for reading and writing sequence data.
+*/
+package bio
+
+import (
+	"io"
+)
+
+/*
+This package is supposed to be empty and only exists to provide a doc string.
+Otherwise its namespace would collide with Go's native IO package.
+*/
+
+type Parser[T any, TH any] interface {
```

Comment:
What are `T` and `TH` here? Some more explanation on the need for an interface and how it works would be very helpful.

Reply:
Types, as are used in generics. I've added in `DataTypes` and `DataTypeHeaders` to be a little more descriptive. While `T` is fairly idiomatic go, `TH` is kinda pushing it, so I think longer descriptions are nice.